### PR TITLE
use using

### PR DESCRIPTION
**Pull request by evilcrash**
some disposables can and should be disposed

### CODE REVIEW

File: Ryujinx.Ui.Common/App/ApplicationLibrary.cs [hunk 1]
```diff
@@ -349,66 +350,72 @@ public void LoadApplications(List<string> appDirs, Language desiredTitleLanguage
...
-                                catch
+                                    catch
                                 {
                                     Logger.Warning?.Print(LogClass.Application, $"The file encountered was not of a valid type. Errored File: {applicationPath}");
 
                                     numApplicationsFound--;
 
                                     continue;
                                 }
+                                }
```
Comment (gdkchan) [inline Ryujinx.Ui.Common/App/ApplicationLibrary.cs:400]: Not properly formatted, also please remove empty line below.

File: Ryujinx.Ava/Common/ApplicationHelper.cs [hunk 2]
```diff
@@ -149,159 +149,162 @@ public static void Initialize(VirtualFileSystem virtualFileSystem, AccountManage
 
             string destination = await folderDialog.ShowAsync(_owner);
 
-            var cancellationToken = new CancellationTokenSource();
-
-            if (!string.IsNullOrWhiteSpace(destination))
+            using (var cancellationToken = new CancellationTokenSource())
             {
-                Thread extractorThread = new(() =>
+
```
Comment (gdkchan) [inline Ryujinx.Ava/Common/ApplicationHelper.cs:154]: Empty line.

File: Ryujinx.HLE/HOS/ApplicationLoader.cs [hunk 3]
```diff
@@ -213,53 +214,54 @@ public static (Nca patch, Nca control) GetGameUpdateData(VirtualFileSystem fileS
 
         public void LoadXci(string xciFile)
         {
-            FileStream file = new FileStream(xciFile, FileMode.Open, FileAccess.Read);
-            Xci xci = new Xci(_device.Configuration.VirtualFileSystem.KeySet, file.AsStorage());
+            using (FileStream file = new FileStream(xciFile, FileMode.Open, FileAccess.Read)) {
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/ApplicationLoader.cs:217]:
```suggestion
            using (FileStream file = new FileStream(xciFile, FileMode.Open, FileAccess.Read))
            {
```

File: Ryujinx.HLE/HOS/ApplicationLoader.cs [hunk 4]
```diff
@@ -683,71 +685,73 @@ public void LoadProgram(string filePath)
...
+                                Logger.Warning?.Print(LogClass.Loader, $"Unsupported ASET header version found \"{asetVersion}\"");
+                            }
                         }
                     }
                 }
             }
             else
             {
-                executable = new NsoExecutable(new LocalStorage(filePath, FileAccess.Read), Path.GetFileNameWithoutExtension(filePath));
+                using var localStorage = new LocalStorage(filePath, fileAccess.Read)
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/ApplicationLoader.cs:753]:
```suggestion
                using var localStorage = new LocalStorage(filePath, fileAccess.Read);
```

File: Ryujinx.HLE/HOS/Horizon.cs [hunk 5]
```diff
@@ -323,7 +323,8 @@ private void InitializeServices()
 
         public void LoadKip(string kipPath)
         {
-            using var kipFile = new SharedRef<IStorage>(new LocalStorage(kipPath, FileAccess.Read));
+            using var localStorage = new LocalStorage(kipPath, FileAccess.Read)
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Horizon.cs:326]:
```suggestion
            using var localStorage = new LocalStorage(kipPath, FileAccess.Read);
```

**Review by gdkchan — COMMENTED** [review]
I'd prefer if scoped usings were used in more places here to avoid all the nesting.

**Comment by AcK77** [comment]
Closed due to inactivity, feel free to fix things and we can reopen it! Thanks for your contribution.

**Pull request closed without merging**